src/vm/amd64/CrtHelpers.asm

```diff
@@ -14,317 +14,74 @@
-        ; the remainder <= 16 
-        movdqu  -16[rcx + r8], xmm0
-        ret
-        
-        ; count > 16 && count <= 128 for mset02
-        align 16
-mset02: movdqu	[rcx], xmm0         
-        movdqu	-16[rcx + r8], xmm0    
-        cmp     r8, 32                 
-        jbe     mset03
-        
-        ; count > 32 && count <= 64
-        movdqu	16[rcx], xmm0
-        movdqu	-32[rcx + r8], xmm0
-        cmp     r8, 64
-        jbe     mset03
-        
-        ; count > 64 && count <= 128
-        movdqu	32[rcx], xmm0
-        movdqu	48[rcx], xmm0
-        movdqu	-48[rcx + r8], xmm0
-        movdqu	-64[rcx + r8], xmm0   
-mset03: ret
- 
-        align 16
-mset04: mov     rax, rcx                ; save dst address
-        test    r8b, 24                 ; and r8b with 0001 1000
-        jz      mset05
-        
-        ; count >= 8 && count <= 16
-        mov     [rcx], rdx        
-        mov     -8[rcx + r8], rdx
-        ret
-
-        align 16
-mset05: test    r8b, 4                  ; and r8b with 0100
-        jz      mset06
-        
-        ; count >= 4 && count < 8
-        mov     [rcx], edx        
-        mov     -4[rcx + r8], edx
-        ret
-        
-        ; count >= 0 && count < 4
-        align 16
-mset06: test    r8b, 1                  ; and r8b with 0001
-        jz      mset07
-        mov     [rcx],dl
-mset07: test    r8b, 2                  ; and r8b with 0010
-        jz      mset08
-        mov     -2[rcx + r8], dx
-mset08: ret
-
 LEAF_END_MARKED JIT_MemSet, _TEXT
 
-;JIT_MemCpy - Copy source buffer to destination buffer
+; void JIT_MemCpy(void* dest, const void* src, size_t count)
 ;
-;Purpose:
-;   JIT_MemCpy() copies a source memory buffer to a destination memory
-;   buffer. This routine recognize overlapping buffers to avoid propogation.
-;   For cases where propogation is not a problem, memcpy() can be used.
+; Purpose:
+;    Copies the values of "count" bytes from the location pointed to
+;    by "src" to the memory block pointed by "dest".
 ;
-;Algorithm:
-;Copy to destination based on count as follow
-;   count [0, 64]: overlap check not needed
-;       count [0, 16]: use 1/2/4/8 bytes width registers  
-;       count [16, 64]: use 16 bytes width registers (XMM) without loop
-;   count [64, upper]: check overlap
-;       non-overlap:
-;           count [64, 512]: use 16 bytes width registers (XMM) with loops, unrolled 4 times
-;           count [512, upper]: use rep movsb
-;       overlap::
-;           use 16 bytes width registers (XMM) with loops to copy from end to beginnig
+; Entry:
+;    RCX: void* dest             - Pointer to the destination array where content is to be copied.
+;    RDX: const void* src        - Pointer to the source of the data to be copied.
+;    R8:  size_t count           - Number of bytes to copy.
 ;
-;Entry:
-;   void *dst = pointer to destination buffer
-;   const void *src = pointer to source buffer
-;   size_t count = number of bytes to copy
+; Exit:
```

Comment:
The declaration of these methods in `jitinterface.h` have them as void returning, rather than as `void*` returning (which is what the CRT implementations do):
https://github.com/dotnet/coreclr/blob/01da3bdd734c887856f89037cd4a3f2b5a072967/src/vm/jitinterface.h#L434